### PR TITLE
Nanopaste and robotic limbs fixes and changes

### PR DESCRIPTION
**Pull request by MacBlaze1**
## About The Pull Request
<details>
<summary>
	Fixes Nanopaste to actually heal robotic limbs again.
</summary>
<hr>
        Fixes nanopaste to work again
<hr>
</details>
## Changelog
:cl:
fix: fixed nanopaste not healing people
/:cl:

### CODE REVIEW

**Comment by ACCount12** [comment]
Looking at the code, this nanopaste thing was intentional - that check you removed just requires a robotic. limb to be screwed open before you can apply nanopaste.

**Comment by TorinoFermic** [comment]
> Looking at the code, this nanopaste thing was intentional - that check you removed just requires a robotic. limb to be screwed open before you can apply nanopaste.

If the check is restored, the message could changed to tell the user how to progress.

File: code/game/objects/items/stacks/nanopaste.dm [hunk 1]
```diff
@@ -33,22 +33,20 @@
...
-						"<span class='notice'>\The [user] applies some nanite paste at[user != M ? " \the [M]'s" : " \the"][S.name] with \the [src].</span>",
-						"<span class='notice'>You apply some nanite paste at [user == M ? "your" : "[M]'s"] [S.name].</span>"
-					)
-				else
-					to_chat(user, SPAN_NOTICE("Nothing to fix here."))
-		else
-			if (can_operate(H))        //Checks if mob is lying down on table for surgery
-				if (do_surgery(H,user,src))
-					return
+		if(BP_IS_ROBOTIC(S))
```
Comment (LethalGhost) [inline code/game/objects/items/stacks/nanopaste.dm]: `S` existence check should not be ommited.

Comment (MacBlaze1) [reply code/game/objects/items/stacks/nanopaste.dm]: Let me add that back

**Comment by MacBlaze1** [comment]
> > Looking at the code, this nanopaste thing was intentional - that check you removed just requires a robotic. limb to be screwed open before you can apply nanopaste.
> 
> If the check is restored, the message could changed to tell the user how to progress.

I tried screwdrivering my arm, didnt work for some reason, I used a crowbar, screwdriver and 1 other device but nothing opened it.

File: code/game/objects/items/stacks/nanopaste.dm [hunk 2]
```diff
@@ -33,20 +33,21 @@
...
-		if (can_operate(H))        //Checks if mob is lying down on table for surgery
-			if (do_surgery(H,user,src))
-				return
+		if(S)
+			if(BP_IS_ROBOTIC(S))
+				if(S.get_damage())
+					user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+					S.heal_damage(15, 15, robo_repair = 1)
+					H.updatehealth()
+					use(1)
```
Comment (LethalGhost) [inline code/game/objects/items/stacks/nanopaste.dm]: It's better to call use before healing like
```
if(use(1))
	S.heal_damage(15, 15, robo_repair = 1)
```
Also `H.updatehealth()` is called within `heal_damage` proc with new organs code.

File: code/game/objects/items/stacks/nanopaste.dm [hunk 3]
```diff
@@ -50,5 +51,4 @@
 			if (can_operate(H))        //Checks if mob is lying down on table for surgery
 				if (do_surgery(H,user,src))
 					return
-			else
-				to_chat(user, SPAN_NOTICE("Nothing to fix in here.")) //back to the original
+
```
Comment (LethalGhost) [inline code/game/objects/items/stacks/nanopaste.dm]: Why that lines is removed?

Comment (MacBlaze1) [reply code/game/objects/items/stacks/nanopaste.dm]: Let me readd that then

**Pull request merged**